quantecon/game_theory/polymatrix_game.py

```diff
@@ -0,0 +1,243 @@
+""""
+Tools for working with games in Polymatrix form.
+
+In a Polymatrix Game, the payoff to a player is the sum
+of their payoffs from bimatrix games against each player.
+i.e. If two opponents deviate, the change in payoff
+is the sum of the changes in payoff of each deviation.
+
+"""
+
+import numpy as np
+from itertools import product
+from collections.abc import Sequence, Mapping
+from typing import Any, TypeAlias, Self
```

Comment:
`Self` does not exist in Python 3.10...

> ```
> E   ImportError: cannot import name 'Self' from 'typing' (/home/runner/miniconda3/envs/qe/lib/python3.10/typing.py)
> ```

(https://github.com/QuantEcon/QuantEcon.py/actions/runs/11947499924/job/33303544452?pr=742#step:7:27)

Reply:
Oh well. I have commented them out.